### PR TITLE
Add ability to manage role grants

### PR DESCRIPTION
**Pull request by wookasz**
**What**
- This only allows granting of roles to users, not roles to roles
- No updates, making changes requires creating a new resource

**Why**
Limiting the functionality this way as allowing both complicates the code unnecessarily and requires handling different returns from the show queries. Granting roles to roles may belong in the grants resource.

A role grant cannot really be updated. If either the name of the role or the name of the user changes, that requires revoking the old role from the old user and granting new role to new user, a.k.a. a brand new grant.

**Plans**
Create
```
------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + snowflake_role_grant.tester
      id:   <computed>
      role: "TESTER"
      user: "LLEMPART"


Plan: 1 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------
```

Update
```
------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

-/+ snowflake_role_grant.tester (new resource required)
      id:   "TESTER-LLEMPART" => <computed> (forces new resource)
      role: "TESTER" => "OTHERTESTER" (forces new resource)
      user: "LLEMPART" => "LLEMPART"


Plan: 1 to add, 0 to change, 1 to destroy.

------------------------------------------------------------------------
```

Delete
```
------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  - snowflake_role_grant.tester


Plan: 0 to add, 0 to change, 1 to destroy.

------------------------------------------------------------------------
```

### CODE REVIEW

File: snowflake/resource_role_grant.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+	stmtSQL := fmt.Sprintf("GRANT ROLE \"%s\" TO USER \"%s\"", role, user)
+
+	log.Println("Executing statement:", stmtSQL)
+
+	if _, err := db.Exec(stmtSQL); err != nil {
+		return err
+	}
+
+	return readRoleGrant(d, meta)
```
Comment (kbial) [inline snowflake/resource_role_grant.go:51]: Do we need to do this? If the query failed the Exec will return an err

File: snowflake/resource_role_grant.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+	_, err := db.Exec(stmtSQL)
+	if err == nil {
+		d.SetId("")
+	}
+
+	return err
+}
+
+func paramsFromRoleGrantID(id string) (role string, user string) {
```
Comment (kbial) [inline snowflake/resource_role_grant.go]: nit role, user string

File: snowflake/provider.go [hunk 3]
```diff
@@ -65,9 +65,10 @@ func Provider() terraform.ResourceProvider {
 		},
 
 		ResourcesMap: map[string]*schema.Resource{
-			"snowflake_warehouse": resourceWarehouse(),
-			"snowflake_database":  resourceDatabase(),
-			"snowflake_user":      resourceUser(),
+			"snowflake_warehouse":  resourceWarehouse(),
+			"snowflake_database":   resourceDatabase(),
+			"snowflake_user":       resourceUser(),
+			"snowflake_role_grant": resourceRoleGrant(),
```
Comment (kbial) [inline snowflake/provider.go:71]: We should probably agree and follow one convention - some resources have methods starting with capital letters, some don't

**Review by kbial — APPROVED** [review]

File: snowflake/resource_role_grant.go (hunk 1 above)
Comment (wookasz) [reply snowflake/resource_role_grant.go:51]: it seems to be common practice across the existing providers to return the value of the read method after creating

File: snowflake/provider.go (hunk 3 above)
Comment (wookasz) [reply snowflake/provider.go:71]: the methods should not be exported, we can go back and fix users

**Pull request merged**